src/NeoSharp.Application/Client/Prompt.cs

```diff
@@ -161,7 +162,7 @@ static Prompt()
             IRpcServer rpcInit,
             IBinarySerializer serializer,
             IBlockchain blockchain,
-            IBlockProcessor blockProcessor,
+            IBlockPool blockPool,
```

Comment:
I am still debating if we want people use IBlockPool directly. On the one side, we don't need pass Size and Capacity through BlockProcessor. On the other side, it is possible to add blocks on the pool that are already in repo (only BlockProcessor check repo before addition on the pool). What're your thoughts?

Reply:
if we take a look on the naming of this objects we have our answer

BlockProcessor - I expect to process blocks
BlookPool - I expect to manage the pool of blocks to be processed

when I want to ask: "How many blocks you have on your pool to process?", who should answer? The BlockProcessor or the BlockPool??

Reply:
If you don't want to give access to the method AddBlock in the prompt object (that make all the sense) we can have two interfaces IBlockPool and IBlockPoolReadOnly

BlookPool implements the interface BlockPoolReadOnly that will contain the query methods and the BlockPool the AddBlock method.

In the end BlookPool class will implement both interfaces but one will be used to interact with the BlockPool and we can track who is using and the ReadOnly is to get information about the state of the BlockPool.